src/crypto/tls/tls_ech.go

```diff
@@ -0,0 +1,816 @@
+// Copyright 2020 Cloudflare, Inc. All rights reserved. Use of this source code
```

Comment:
Perhaps "ech.go" would be better than "tls_ech.go?" Here and below.

Reply:
makes sense, go for it!